### PR TITLE
Uidesign

### PR DESCRIPTION
**Pull request by arushigarg18**
Added the code for basic ui prototype

### CODE REVIEW

File: experimental/css.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  .btn-add {        
+    display: inline;
+    float: right; 
+    height: 20px; 
+    line-height: 20px;
+    min-width: 40px; 
+    max-width: 40px; 
+  }
+  
+  .btn-delete1 {
```
Comment (SowmithManepalli) [inline experimental/css.html]: Add a description to the css class. how is it different from btn-delete2.

File: experimental/css.html [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    padding: 5px;
+    padding-bottom: 20px;   
+  }
+   
+  .box-slice {
+    border: 1px solid #e5e5e5;
+    padding: 10px;
+  }
+  
+  .drp1 {
```
Comment (SowmithManepalli) [inline experimental/css.html]: nit: Add a description to the css class. how is it different from btn-delete2, or can modify the class name to be more obvious.

File: experimental/css.html (hunk 1 above)
Comment (arushigarg18) [reply experimental/css.html]: In the new spreadsheet I have removed this classes and added classes for icons with appropriate names.

File: experimental/css.html (hunk 2 above)
Comment (arushigarg18) [reply experimental/css.html]: Done

File: experimental/javascript.html [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                    '</select>'+
+                  '</div>'+
+                '</div>';
+      addElement('granularity','div',"grn",html);   //parentId, elementTag, elementId, html
+      button.classList.remove("share","btn-add");
+      button.classList.add("red","btn-delete2");
+      button.textContent="DELETE";
+    }
+    else{
+      console.log("removing");
```
Comment (SowmithManepalli) [inline experimental/javascript.html]: Elaborate the log message or remove the statement.

File: experimental/javascript.html [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                '<button name="delete-filter" class="red btn-del-filter" onclick="removeElement(\''+sliceId+'\')">x</button>'+
+                '<br>'+
+                '<div id="part1-slice" class="block">'+
+                  '<label for="slice-col'+ids+'" class="inline">Slice Column Name:&ensp;</label>'+
+                  '<input type="text" name="slice-col" id="slice-col'+ids+'" class="width90" placeholder="column">'+
+                '</div>'+
+                '<div id="part2-slice" class="block">'+
+                  '<label for="slice-op'+ids+'" class="inline">Filter Operation:&ensp;</label>'+
+                  '<select id="slice-op'+ids+'" name="slice-op" class="width100">'+
+                    '<option>Equal to</option>'+
```
Comment (SowmithManepalli) [inline experimental/javascript.html]: can we have a helper function that does the same thing. can just call it everywhere you generate options.

File: experimental/javascript.html [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    else{
+      console.log("removing");
+      removeElement("grn");
+      button.classList.remove("red","btn-delete2");
+      button.classList.add("share","btn-add");
+      button.textContent="ADD";
+    }
+  }
+    
+  function addElement(parentId, elementTag, elementId, html) {
```
Comment (SowmithManepalli) [inline experimental/javascript.html]: Can move these kind of helper functions to separate file called helper.gs.

File: experimental/javascript.html [hunk 6]
```diff
@@ -0,0 +1,161 @@
+<script>
+
```
Comment (SowmithManepalli) [inline experimental/javascript.html]: Can you add a comment on what the functions of this file (javascript.html) accomplish. I can understand that this is javascript responsible in manipulating the UI sidebar. But lets also add that here.

File: experimental/javascript.html (hunk 3 above)
Comment (arushigarg18) [reply experimental/javascript.html]: Okay, will do

File: experimental/javascript.html (hunk 6 above)
Comment (arushigarg18) [reply experimental/javascript.html]: Okay, I will add it in brief

File: experimental/javascript.html (hunk 5 above)
Comment (arushigarg18) [reply experimental/javascript.html]: Will do that

File: experimental/javascript.html (hunk 4 above)
Comment (arushigarg18) [reply experimental/javascript.html]: Yeah, I will write a function for it

File: Add-On UI/javascript/aspects/date_range.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
+<!-- Javascript Functions for adding date-range field -->
+
+<script>
+
+  function funcAddDate(icon) {
+  
+    var text=icon.innerHTML;
+    
+    // User clicks add button
+    if(text === "add_circle") {
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/aspects/date_range.html:10]: what does "add_circle" signify here.

File: Add-On UI/javascript/submit_form.html [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+<script>
+
+  var form = document.getElementById("form");  //fetching form
+  var done = document.getElementById("done");  //fetching the submit button
+  
+  // Function called when user clicks the Compute (submit) button
+  function submitForm(intentName,event) {
+  
+    event.preventDefault();
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/submit_form.html:12]: add a comment on what the line accomplishes. Also add what is the default action that you are cancelling , and why if it is not too complex.

File: Add-On UI/javascript/submit_form.html [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+    var slices = fetchSlice();  //1D array of objects containing slicing information
+    
+    //date
+    var dateRange = fetchDate();  //date column, start date, end date
+    
+    //granularity
+    var timeGranularity = fetchGranularity();  //contains time granularity value
+    
+    //output sheet name
+    //var outputSheet = fetchOutputSheetName();  //for extracting output sheet name
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/submit_form.html:49]: are we going to take output sheet name as part of the form , if yes delete the line and add a todo on yourself.

File: Add-On UI/javascript/submit_form.html [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+    var outputSheet = 'Output Sheet';            //setting output sheet default name
+    
+    google.script.run
+          .withSuccessHandler(printOutput)       //call printOutput when server-side function returns successfully
+          .createJsonObject(                     //server side function
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/submit_form.html:54]: Does createJsonObject also make the backend function call to gcp ? If yes may be modify the function name or add a comment.

File: Add-On UI/javascript/submit_form.html (hunk 9 above)
Comment (SowmithManepalli) [reply Add-On UI/javascript/submit_form.html:49]: Something like. 
// TODO(arushigarg) fetch outputname from user.

File: Add-On UI/scripts/process_query.js [hunk 11]
```diff
@@ -0,0 +1,236 @@
+//json object that contains all aspects and table's data
+var JsonObj = {
+  'intent': 'null',
+  'table': 'null',
+  'rowRange': 'null',
```
Comment (SowmithManepalli) [inline Add-On UI/scripts/process_query.js:5]: rowRange seems confusing , can we table-metadata or something like that

File: Add-On UI/scripts/process_query.js [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+  'metric': 'null',
+  'dimensions': 'null',
+  'summaryOperator': 'null',
+  'isAsc': 'null',
+  'k': 'null',
```
Comment (SowmithManepalli) [inline Add-On UI/scripts/process_query.js:10]: 'k' also doesnt really tell a lot about the argument.

File: Add-On UI/scripts/process_query.js [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+  var objRowRange = {
+      'header' : 1,
+      'rowStart': 2,
+      'rowEnd': 2
+  }
+
+  //if range is not specified we consider entire data 
+  //else we use the range provided
+
+  if(tableData[0] === '') {
```
Comment (SowmithManepalli) [inline Add-On UI/scripts/process_query.js:121]: Less discuss offline in chat on what is exactly happening here.  how is  if(tableData[0] === '') checking that range is not specified, looks like i m missing something here.

File: Add-On UI/javascript/submit_form.html [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+            dateRange,
+            timeGranularity,
+            outputSheet);
+ 
+  }
+  
+  //fetching table range and header row for table
+  function fetchTable() {
+  
+    var tableData = [];
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/submit_form.html:72]: Add a comment on what exactly does table data consist of.

File: Add-On UI/javascript/submit_form.html [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+      limit = form.elements["limit"].value;
+      if(limit === "")
+        limit = 5000000;
+      else 
+        limit = parseInt(limit);
+    }
+      
+    //limit is an integer with a default max value of 5000000 
+    //since 5000000 is the maximum cell limit in sheets
+    console.log(limit);
```
Comment (SowmithManepalli) [inline Add-On UI/javascript/submit_form.html:136]: delete

File: Add-On UI/scripts/process_query.js [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+  };
+    
+  var response= UrlFetchApp.fetch(
+                  "https://us-central1-mocha-interns.cloudfunctions.net/testfunction",
+                  options);
+
+  var outputTable = JSON.parse(response.getContentText());
+
+  //printing output received from gcp python function
+  Logger.log(outputTable);
```
Comment (SowmithManepalli) [inline Add-On UI/scripts/process_query.js:168]: Add a string message for understanding logs better.  Logger.log (Output received from python function). Do this for the other logger.logs also.

File: Add-On UI/javascript/aspects/date_range.html (hunk 7 above)
Comment (arushigarg18) [reply Add-On UI/javascript/aspects/date_range.html:10]: This is the name of icon being used.

File: Add-On UI/javascript/submit_form.html (hunk 9 above)
Comment (arushigarg18) [reply Add-On UI/javascript/submit_form.html:49]: Done

File: Add-On UI/javascript/submit_form.html (hunk 10 above)
Comment (arushigarg18) [reply Add-On UI/javascript/submit_form.html:54]: Yeah, it does. I will change the function name.

File: Add-On UI/javascript/submit_form.html (hunk 14 above)
Comment (arushigarg18) [reply Add-On UI/javascript/submit_form.html:72]: Yeah, I have added a comment above the function. I will write range in A1 notation for more clarity.

File: Add-On UI/javascript/submit_form.html (hunk 15 above)
Comment (arushigarg18) [reply Add-On UI/javascript/submit_form.html:136]: what to delete ?

File: Add-On UI/scripts/process_query.js (hunk 11 above)
Comment (arushigarg18) [reply Add-On UI/scripts/process_query.js:5]: Yes, I will change it.

File: Add-On UI/scripts/process_query.js (hunk 12 above)
Comment (arushigarg18) [reply Add-On UI/scripts/process_query.js:10]: Should I write limit or something other name ?

File: Add-On UI/scripts/process_query.js (hunk 16 above)
Comment (arushigarg18) [reply Add-On UI/scripts/process_query.js:168]: Okay

File: Add-On UI/javascript/submit_form.html (hunk 8 above)
Comment (arushigarg18) [reply Add-On UI/javascript/submit_form.html:12]: Okay, I will add a short explanation

**Pull request merged**